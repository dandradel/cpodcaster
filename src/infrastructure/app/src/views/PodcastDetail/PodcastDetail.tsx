import { Box } from '@mui/material'
import React, { ReactElement } from 'react'
import Episodes from '../../components/Episodes/Episodes'
import Error from '../../components/Error/Error'
import PodcastCard from '../../components/PodcastCard/PodcastCard'
import '../../components/PodcastCover/PodcastCover.styles.css'
import useError from '../../shared/hooks/useError'
import { podcastDetailBoxStyles } from './styles'
import usePodcastById from '../../shared/hooks/usePodcastById'
import Skeleton from '../../components/Skeleton/Skeleton'

function PodcastDetail(): ReactElement {
  const { podcastCardProps, episodes } = usePodcastById()
  const { error } = useError()

  if (error) {
    return <Error />
  }

  if (!podcastCardProps || !episodes) {
    return <Skeleton />
  }

  return (
    <>
      <Box sx={podcastDetailBoxStyles}>
        <PodcastCard podcastCard={podcastCardProps} />
        <Episodes episodes={episodes} />
      </Box>
    </>
  )
}

export default PodcastDetail
