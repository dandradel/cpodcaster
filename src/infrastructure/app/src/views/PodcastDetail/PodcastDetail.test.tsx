import React from 'react'
import '@testing-library/jest-dom'
import { cleanup, renderWithProviders, screen } from '../../shared/test/test-utils'
import PodcastDetail from './PodcastDetail'
import episodeMock from '../../shared/__mocks__/episodes.json'

jest.mock('../../shared/hooks/usePodcastById', () => ({
  __esModule: true,
  default: () => ({
    podcastCardProps: {
      podcastImage: '1',
      podcastName: 'test',
      podcastAuthor: 'test',
      podcastDescription: 'test',
    },
    episodes: episodeMock,
  }),
}))

describe('PodcastDetail', () => {
  afterEach(cleanup)

  test('renders PodcastDetail', () => {
    const container = renderWithProviders(<PodcastDetail />)
    expect(container).toMatchSnapshot()
  })

  test('renders PodcastDetail with PodcastCard', () => {
    renderWithProviders(<PodcastDetail />)
    const podcastCard = screen.getByTestId('podcast-wrapper-testid')
    expect(podcastCard).toBeInTheDocument()
  })

  test('renders PodcastDetail with Episodes', () => {
    renderWithProviders(<PodcastDetail />)
    const episodes = screen.getByTestId('episodes-wrapper-testid')
    expect(episodes).toBeInTheDocument()
  })
})
