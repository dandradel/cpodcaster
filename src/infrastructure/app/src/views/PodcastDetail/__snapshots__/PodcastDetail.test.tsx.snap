// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PodcastDetail renders PodcastDetail 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-70hkv0"
      >
        <div
          class="MuiBox-root css-6m2utx"
          data-testid="podcast-wrapper-testid"
        >
          <div
            class="MuiBox-root css-1ia54b9"
          >
            <img
              class="podcastCard__image"
              data-testid="podcast-image-testid"
              src="1"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-1d8isoc"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
              data-testid="podcast-name-testid"
            >
              test
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
              data-testid="podcast-author-testid"
            >
              by 
              test
            </h6>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-tcr4li"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
              data-testid="podcast-description-testid"
            >
              Description:
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
              data-testid="podcast-description1-testid"
            >
              test
            </h6>
          </div>
        </div>
        <div
          class="MuiBox-root css-1ixmh22"
          data-testid="episodes-wrapper-testid"
        >
          <div
            class="MuiBox-root css-4zyzsf"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
              data-testid="episodes-number-testid"
            >
              Episodes: 
              1
            </h6>
          </div>
          <div
            class="MuiBox-root css-13z1pn"
          >
            <div
              class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
            >
              <table
                class="MuiTable-root css-rqglhn-MuiTable-root"
                data-testid="table-episodes-testid"
              >
                <thead
                  class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-11sptls-MuiTableCell-root"
                      scope="col"
                    >
                      Title
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                      scope="col"
                    >
                      Date
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                      scope="col"
                    >
                      Duration
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                >
                  <tr
                    class="MuiTableRow-root css-1kapz33-MuiTableRow-root"
                    data-testid="episodes-tableRow-testid"
                  >
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-siow7j-MuiTableCell-root"
                    >
                      Episode 1
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                    >
                      1/1/2020
                    </td>
                    <td
                      class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                    >
                      25:25
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-70hkv0"
    >
      <div
        class="MuiBox-root css-6m2utx"
        data-testid="podcast-wrapper-testid"
      >
        <div
          class="MuiBox-root css-1ia54b9"
        >
          <img
            class="podcastCard__image"
            data-testid="podcast-image-testid"
            src="1"
          />
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1d8isoc"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom css-165m6tq-MuiTypography-root"
            data-testid="podcast-name-testid"
          >
            test
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-t9vmtp-MuiTypography-root"
            data-testid="podcast-author-testid"
          >
            by 
            test
          </h6>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-tcr4li"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-16yea3k-MuiTypography-root"
            data-testid="podcast-description-testid"
          >
            Description:
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-4qfdts-MuiTypography-root"
            data-testid="podcast-description1-testid"
          >
            test
          </h6>
        </div>
      </div>
      <div
        class="MuiBox-root css-1ixmh22"
        data-testid="episodes-wrapper-testid"
      >
        <div
          class="MuiBox-root css-4zyzsf"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
            data-testid="episodes-number-testid"
          >
            Episodes: 
            1
          </h6>
        </div>
        <div
          class="MuiBox-root css-13z1pn"
        >
          <div
            class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
              data-testid="table-episodes-testid"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-11sptls-MuiTableCell-root"
                    scope="col"
                  >
                    Title
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                    scope="col"
                  >
                    Date
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                    scope="col"
                  >
                    Duration
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root css-1kapz33-MuiTableRow-root"
                  data-testid="episodes-tableRow-testid"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-siow7j-MuiTableCell-root"
                  >
                    Episode 1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                  >
                    1/1/2020
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium css-177gid-MuiTableCell-root"
                  >
                    25:25
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
